### PR TITLE
govivace_stt

### PR DESCRIPTION
**Pull request by JarbasAl**
## Description
adds goVivace STT option

## How to test
- adquire key
- update config
- use mycroft by voice

## Contributor license agreement signed?
CLA [yes] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/stt/__init__.py [hunk 1]
```diff
@@ -201,6 +201,29 @@ def execute(self, audio, language=None):
...
+        self.default_uri = "https://services.govivace.com:49149/telephony"
+
+        if not self.lang.startswith("en") and not self.lang.startswith("es"):
+            LOG.error("GoVivace STT only supports english and spanish")
+            raise NotImplementedError
+
+    def execute(self, audio, language=None):
+        url = self.config.get("uri", self.default_uri) + "?key=" + self.token + \
+              "&action=find&format=8K_PCM16&validation_string="
+        response = put(url,
```
Comment (forslund) [inline mycroft/stt/__init__.py]: `put` should be `requests.put`

File: mycroft/stt/__init__.py [hunk 2]
```diff
@@ -201,6 +201,29 @@ def execute(self, audio, language=None):
...
+                       data=audio.get_wav_data())
+        return self.get_response(response)
+
+    def get_response(self, response):
+        try:
+            return response.json()["result"]["hypotheses"][0]["transcript"]
+        except:
+            return None
```
Comment (forslund) [inline mycroft/stt/__init__.py]: This except catch all is not helpful, it eats any indication of what is wrong and returns `None` which raises an error in the listener instead.

File: mycroft/configuration/mycroft.conf [hunk 3]
```diff
@@ -209,14 +209,18 @@
...
     "module": "mycroft"
     // "deepspeech_server": {
     //   "uri": "http://localhost:8080/stt"
     // },
     // "kaldi": {
     //   "uri": "http://localhost:8080/client/dynamic/recognize"
     // },
+    //"govivace": {
+    //   "uri": "https://services.govivace.com:49149/telephony",
+    //   "token": "xxxxx"
```
Comment (forslund) [inline mycroft/configuration/mycroft.conf]: Should be
```json
    "credential": {
      "token": "xxxxxx"
      }
```

File: mycroft/stt/__init__.py (hunk 2 above)
Comment (JarbasAl) [reply mycroft/stt/__init__.py]: kaldi does it, aren't all STT engines supposed to return None if speech was not recognized?

**Comment by forslund** [comment]
I think I've found the issue. Mycroft records at a rate of 16kHz

```python
LOG.info('Audio: {}, {}'.format(audio.sample_width, audio.sample_rate))
```
07:25:06.729 - mycroft.stt:execute:218 - INFO - Audio: 2, 16000

So when passing the wav_data to the service you'll need to convert it to the expected 8kHz rate:

```python
response = put(url, data=audio.get_wav_data(convert_rate=8000))
```

**Comment by forslund** [comment]
Nice! Seems to work fine now.

**Pull request merged**